src/tests/build.proj

```diff
@@ -51,5 +54,142 @@
+
+  <Target Name="BuildManagedTestGroups" DependsOnTargets="RestorePackages;ResolveDisabledProjects">
+    <Message Importance="High" Text="$(MsgPrefix)Building managed test components" />
+    <!-- Execute msbuild test build in stages - workaround for excessive data retention in MSBuild ConfigCache -->
+    <!-- See https://github.com/Microsoft/msbuild/issues/2993 -->
+
+    <!-- We need to build group #1 manually as it doesn't have a _GroupStartsWith item associated with it, see the comment in Common\dirs.proj -->
+    <MSBuild Projects="$(MSBuildThisFileFullPath)" Targets="BuildManagedTestGroup;CopyNativeTestBinaries" Properties="__TestGroupToBuild=1;__SkipRestorePackages=1" />
+    <MSBuild Projects="$(MSBuildThisFileFullPath)" Targets="BuildManagedTestGroup;CopyNativeTestBinaries" Properties="__TestGroupToBuild=%(_GroupStartsWith.GroupNumber);__SkipRestorePackages=1" />
+  </Target>
+
+  <Target Name="BuildManagedTestGroup"
+      DependsOnTargets="ResolveDisabledProjects"
+      Condition="'$(__SkipManaged)' != '1' and '$(__CopyNativeTestBinaries)' != '1'" >
+
+    <PropertyGroup>
+      <GroupBuildCmd>$(DotNetCli) msbuild</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) $(MSBuildThisFileFullPath)</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) /t:Build</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:TargetArchitecture=$(TargetArchitecture)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:Configuration=$(Configuration)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:LibrariesConfiguration=$(LibrariesConfiguration)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:TargetOS=$(TargetOS)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:RuntimeFlavor=$(RuntimeFlavor)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:RuntimeVariant=$(RuntimeVariant)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:CLRTestBuildAllTargets=$(CLRTestBuildAllTargets)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:__TestGroupToBuild=$(__TestGroupToBuild)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:__SkipRestorePackages=1"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) /nodeReuse:false</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) /maxcpucount</GroupBuildCmd>
+    </PropertyGroup>
+
+    <Message Importance="High" Text="$(MsgPrefix)Building managed test group $(__TestGroupToBuild): $(GroupBuildCmd)" />
+
+    <Exec Command="$(GroupBuildCmd)" />
+  </Target>
+
+  <!-- In split pipeline mode (in the lab) we're using the native component copying step to generate the test execution scripts -->
+  <Target Name="CopyNativeTestBinaries"
+      DependsOnTargets="ResolveDisabledProjects"
+      Condition="'$(__CopyNativeTestBinaries)' == '1'">
+
+    <Message Importance="High" Text="$(MsgPrefix)Copying native binaries for test group $(__TestGroupToBuild)" />
+    <CallTarget Targets="CopyAllNativeTestProjectBinaries" />
```

Comment:
`<CallTarget>` is generally not considered good practice in MSBuild. Would it be possible to chain this in a different way (either recursive `<MSBuild />` or with target dependencies)?

Reply:
Fixed in latest commit.